kong/conf_loader.lua

```diff
@@ -357,6 +389,99 @@ local function check_and_infer(conf, opts)
     end
   end
 
+  for _, name in ipairs({
+    "cassandra_consistency_proxy",
+    "cassandra_consistency_admin"
+  }) do
+    local read_consistency
+    local write_consistency
+    local option = conf[name]
+    local number_of_values = option and #option or 0
+    if number_of_values > 2 then
+      if is_cassandra then
+        errors[#errors + 1] = fmt("%s has an invalid value: '%s' (%s)", name,
+                                  concat(option, ", "),
+                                  "specify at most two values, e.g. " ..
+                                  "'read:ONE, write:ONE', or just one " ..
+                                  "for both e.g. 'QUORUM'")
+      end
+
+    elseif number_of_values == 2 then
+      read_consistency = CASSANDRA_READ_CONSISTENCY[option[1]] or
+                         CASSANDRA_READ_CONSISTENCY[option[2]]
+
+      write_consistency = CASSANDRA_WRITE_CONSISTENCY[option[1]] or
+                          CASSANDRA_WRITE_CONSISTENCY[option[2]]
+
+      if not read_consistency or not write_consistency and is_cassandra then
+        errors[#errors + 1] = fmt("%s has an invalid value: '%s' (read:%s, " ..
+                                  "write:%s)", name, concat(option, ", "),
+                                  concat(CASSANDRA_CONSISTENCY, ", read:"),
+                                  concat(CASSANDRA_CONSISTENCY, ", write:"))
+      end
+
+    elseif number_of_values == 1 then
+      local found
+      for _, name in ipairs(CASSANDRA_CONSISTENCY) do
+        if name == option[1] then
+          found = true
+        end
+      end
+
+      if not found then
+        if is_cassandra then
+          errors[#errors + 1] = fmt("%s has an invalid value: '%s' (%s)", name,
+                                    option[1],
+                                    concat(CASSANDRA_CONSISTENCY, ", "))
+        end
+
+      else
+        read_consistency  = option[1]
+        write_consistency = option[1]
+      end
+    end
+
+    if not read_consistency then
+      read_consistency  = conf.cassandra_consistency
+    end
+
+    if not write_consistency then
+      write_consistency = conf.cassandra_consistency
+    end
+
+    if read_consistency and write_consistency then
+      if opts.starting and is_cassandra and is_dc_aware then
+        local suggestion
+        if read_consistency == "QUORUM" or
+           read_consistency  == "ONE" then
+          if write_consistency == "QUORUM" or
+             write_consistency == "ONE" then
```

Comment:
This logic should be "if not contains('LOCAL')". Currently this skips other consistency settings such as `TWO`, `THREE`, or even `ALL`.